src/web3/contracts.ts

```diff
@@ -109,6 +116,30 @@ async function waitForContractKit(tries: number) {
   return contractKit
 }
 
+export function* getViemWallet(chain: Chain) {
+  if (viemWallets.has(chain)) {
+    return viemWallets.get(chain) as ViemWallet
+  }
+  const walletAddress = yield* select(walletAddressSelector)
+  const mnemonic = yield* call(getStoredMnemonic, walletAddress)
+  if (!mnemonic) {
+    throw new Error('No mnemonic found')
+  }
+  const keys = yield* call(generateKeysFromMnemonic, mnemonic)
+  const privateKey = keys.privateKey
+  if (!privateKey) {
+    throw new Error('Failed to convert mnemonic to hex')
+  }
+  const wallet = getLockableViemWallet(
+    keychainLock,
+    chain,
+    normalizeAddressWith0x(privateKey) as Address
+  )
```

Comment:
This means that a viem wallet instance is created only when we have a private key right? Would we ever need a wallet instance when the account is not unlocked?

Reply:
Correct.
I don't see anything in the import or create wallet flows that would require a wallet before its unlocked. 

The only other instance is when we load data from the Keychain. That is currently logic that lives in KeychainWallet and we will eventually have to de-couple from contractkit. When we do that de-coupling we shouldn't need an actual wallet before unlock, just some construct to track that a specific address has been loaded from Keychain.